### PR TITLE
Add metrics to prometheus

### LINKED ISSUE #2293: Add metrics to prometheus
Hope add some metrics to prometheus metrics.

Add subscription msgBacklog
Add subscription blockedSubscriptionOnUnackedMsgs
Add consumer blockedSubscriptionOnUnackedMsgs
Add subscription msgRateRedeliver
Add consumer msgRateRedeliver
Add subscription unackedMessages
Add consumer unackedMessages

### PR DESCRIPTION
**Pull request by codelipenghui**
### Motivation

Fixes #2293 

### Changes

Add subscription msgBacklog
Add subscription blockedSubscriptionOnUnackedMsgs
Add consumer blockedSubscriptionOnUnackedMsgs
Add subscription msgRateRedeliver
Add consumer msgRateRedeliver
Add subscription unackedMessages
Add consumer unackedMessages

Should care about memory leak by larger metrics?

### CODE REVIEW

**Comment by codelipenghui** [comment]
@jiazhai 帮我review一下
有个问题  增加了一些指标，指标太多可能导致内存溢出。是否需要加个配置，默认将订阅和消费者的指标关闭

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/stats/prometheus/TopicStats.java [hunk 1]
```diff
@@ -6,9 +6,9 @@
  * to you under the Apache License, Version 2.0 (the
  * "License"); you may not use this file except in compliance
  * with the License.  You may obtain a copy of the License at
- *
- *   http://www.apache.org/licenses/LICENSE-2.0
- *
+ * <p>
```
Comment (jiazhai) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/stats/prometheus/TopicStats.java]: Seems this change will make the jenkins fail to recognise the  header. Please change it back.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/stats/prometheus/NamespaceStatsAggregator.java [hunk 2]
```diff
@@ -6,9 +6,9 @@
  * to you under the Apache License, Version 2.0 (the
  * "License"); you may not use this file except in compliance
  * with the License.  You may obtain a copy of the License at
- *
- *   http://www.apache.org/licenses/LICENSE-2.0
- *
+ * <p>
```
Comment (jiazhai) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/stats/prometheus/NamespaceStatsAggregator.java]: Seems this change will make the jenkins fail to recognise the header. Please change it back.

**Comment by jiazhai** [comment]
@codelipenghui, Agree your comments, It is a good idea to have a config to control these stats for  subscriptions and consumers.

**Comment by jiazhai** [comment]
@sijie @merlimat  Would you please help review this?

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/stats/prometheus/NamespaceStatsAggregator.java (hunk 2 above)
Comment (sijie) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/stats/prometheus/NamespaceStatsAggregator.java]: @codelipenghui you can run "mvn license:format" to format the files with proper apache license headers.

**Comment by codelipenghui** [comment]
Move to [#2299](https://github.com/apache/incubator-pulsar/pull/2299). so close this pull request

**Pull request closed without merging**